Src/FluentAssertions/Specialized/AsyncFunctionAssertions.cs

```diff
@@ -132,21 +132,21 @@ public AsyncFunctionAssertions(Func<TTask> subject, IExtractExceptions extractor
             .BecauseOf(because, becauseArgs)
             .FailWith("Expected {context} to throw exactly {0}{reason}, but found <null>.", expectedType);
 
-        if (!success)
+        if (success)
         {
-            return new ExceptionAssertions<TException>(Array.Empty<TException>());
-        }
+            Exception exception = await InvokeWithInterceptionAsync(Subject);
 
-        Exception exception = await InvokeWithInterceptionAsync(Subject);
+            Execute.Assertion
+                .ForCondition(exception is not null)
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected {0}{reason}, but no exception was thrown.", expectedType);
 
-        Execute.Assertion
-            .ForCondition(exception is not null)
-            .BecauseOf(because, becauseArgs)
-            .FailWith("Expected {0}{reason}, but no exception was thrown.", expectedType);
+            exception.Should().BeOfType(expectedType, because, becauseArgs);
 
-        exception.Should().BeOfType(expectedType, because, becauseArgs);
+            return new ExceptionAssertions<TException>(new[] { exception as TException });
```

Comment:
Somehow, this is an early return too, isn't it?
In this case I do not understand, why this is better read than before.
If there is only one return, ok. But here?

Reply:
Every method that has N returns, will have N-1 that are earlier.
You can search for `^ *return (.*\r\n){1,3} *return` to find existing methods that have closely located return statements.
So it's not a religious "you cannot have multiple return statements".
Also preference change over time, so you'll most likely be able to find a spot where the code violate our currently desired style.

It's a preference of Dennis that he chose for this project and I respect that.
Please, let's not have a lengthy argument about what the universally best style is (spoiler: it does not exist)